spec-proposal/scalars-implementation-guide.md

```diff
@@ -0,0 +1,112 @@
+
+## Result Coercion
+Result coercion is implemented as a single function per Scalar type called "coerceResult". The input is a "raw result value" and the output a "coerced result value". The raw result value is the output of a previously invoked resolver function, which is then coerced by invoking `coerceResult`.
+
+The coerced result value is a leaf value of the overall execution result which is serialized to JSON. The serialized JSON value can be observed and a GraphQL Scalar spec must clearly define the possible JSON values that a Scalar type can produce.
+
+## Input Coercion
+Input coercion deals with three different kind of values: literal, raw input value and coerced input value.
+
+### Literal
+A literal is an element of the GraphQL language representing a fixed value. Examples are `"foo"` or `123` for `StringValue` and `IntValue` literals. Literals are used for:
+
+1. argument values for fields or directives
+
+1. argument default values for fields or directives
+
+1. variable default values of a GraphQL operation
+
+### Raw Input Value
+A raw input value is one of the following:
+
+1. variable value for a GraphQL request
+
+1. programmatically provided default argument value for a field or directive
+
+1. programmatically provided default value for an input object field
+
+1. programmatically provided value for a directive argument in a SDL
+
+Not every GraphQL implementation may offer a programmatic way to provide these input values, but every implementation needs to allow for raw input values offered as variable values for a request.
+
+The details of a raw input value is implementation specific. In JavaScript it might a Map, in Java it might a object. The only way to observe a raw input value is to through the not yet deserialized JSON value.
+
+### Coerced Input Value
+A coerced input value is a value that is ready for consumption by a resolver. It is coerced either from a literal or raw input value.
+
+Input coercion is implemented as a set of three functions `parseLiteral`, `parseRawInputValue` and `rawInputValueToLiteral`.
+
+![image](Coercion.png)
+
+### Parse Literal
+`parseLiteral` takes a GraphQL language literal as input and produces a coerced input value as output.
+
+### Parse Raw Input Value
+`parseRawInputValue` takes a "raw input value" as input and produces a "coerced input value".
+
+### Raw Input Value to Literal
+`rawInputValueToLiteral` takes a raw input value as input and produces a GraphQL language literal.
```

Comment:
`rawInputValueToLiteral` throught the text but `rawInputVariableToLiteral` on the picture

Reply:
Great pickup, thanks

Reply:
FYI we have continued working on this in another repo, under the GraphQL Foundation https://github.com/graphql/graphql-scalars